### PR TITLE
Better default Random Forest parameters; re-enable hyper-param optimization

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

**Comment by acrellin** [comment]
Won't pass until https://github.com/cesium-ml/cesium/pull/224 is merged.

File: cesium_app/ext/sklearn_models.py [hunk 1]
```diff
@@ -121,6 +119,13 @@ def check_model_param_types(model_type, model_params, all_as_lists=False):
         Boolean indicating whether `model_params` values are wrapped in lists,
         as in the case of parameter grids for optimization.
 
+    Returns
+    -------
+    (dict, dict) tuple
+        Returns a tuple of two dictionaries, the first of which contain those
+        hyper-parameters determined to passed into model constructor as-is, and
```
Comment (stefanv) [inline cesium_app/ext/sklearn_models.py]: check this sentence

Comment (stefanv) [reply cesium_app/ext/sklearn_models.py]: Also, describe what the dictionaries contain as values

File: cesium_app/ext/sklearn_models.py [hunk 2]
```diff
@@ -145,13 +150,12 @@ def verify_type(value, vtype):
             Check that value is one of these types.
 
         """
-        values = [value]
+        vtypes = set(make_list(vtype))
+        values = make_list(value) if list not in vtypes else [value]
```
Comment (stefanv) [inline cesium_app/ext/sklearn_models.py:203]: What does this do?

File: cesium_app/ext/sklearn_models.py [hunk 3]
```diff
@@ -172,7 +179,16 @@ def verify_type(value, vtype):
 
         try:
             verify_type(param_value, required_type)
+            if (make_list(param_value) == param_value and list not in
```
Comment (stefanv) [inline cesium_app/ext/sklearn_models.py:233]: Here's similar logic---feels like the kind of thing that should become a small utility function, just to benefit readability?

File: public/scripts/Models.jsx [hunk 4]
```diff
@@ -87,22 +87,34 @@ NewModelForm.propTypes = {
...
 
-  const paramDefaults = {};
-  currentModel.params.map((param) => {
-    paramDefaults[param.name] = (param.default === null) ? "None" : param.default;
-  });
-
   const firstFeatureSet = state.featuresets[0];
   const firstFeatureSetID = firstFeatureSet ? firstFeatureSet.id : "";
 
+  const selectedFsetID = $try(() => formState.featureSet.value) || 0;
```
Comment (stefanv) [inline public/scripts/Models.jsx]: Is this `|| 0` correct?

File: public/scripts/Models.jsx [hunk 5]
```diff
@@ -87,22 +87,34 @@ NewModelForm.propTypes = {
...
+  const selectedFset = state.featuresets.filter(fset => (fset.id == selectedFsetID));
+  const nFeats = selectedFset.length > 0 ? selectedFset[0].features_list.length : null;
+
+  let paramDefaults = {};
```
Comment (stefanv) [inline public/scripts/Models.jsx]: Why is this whole section necessary?

**Review by stefanv — COMMENTED** [review]
Sorry, forgot to submit this earlier; some comments may

**Comment by acrellin** [comment]
@stefanv none of the Javascript changes exist anymore...😉

**Comment by stefanv** [comment]
\o/  Yeah, sorry, those were old comments lying around.

File: cesium_app/ext/sklearn_models.py (hunk 3 above)
Comment (acrellin) [reply cesium_app/ext/sklearn_models.py:233]: Some similar logic, but different enough that they couldn't both use the same utility function

**Comment by acrellin** [comment]
This should now pass when https://github.com/cesium-ml/cesium/pull/232 gets merged.

**Pull request merged**